### PR TITLE
Refactoring of drawGraph, and added graphUtils.c & .h.

### PR DESCRIPTION
**Pull request by chrnigi**
### General Changes
Changed all instances of the typedef'd `tm` to `struct tm` in all in drawGraph.c, drawGraph.h, graphUtils.c and graphUtils.h (I might have missed some.) We should maybe change this for the rest of the source code aswell.

### Changes to `graph_exec`
`graph_exec` now takes the individual fields of the GraphParams struct as parameters, so as to more easily add possible parameters to this.

### New files: graphUtils.c and graphUtils.h
The new functions defined and declared in these are:

`time_input`:
 Returns a struct tm from user input.

`wcstrftime`: 
Takes 4 parameters,  `wchar_t * dest, size_t max_len, const char *format, const struct tm *timeptr`.  The function writes a `strftime` style format-string to a wchar_t array of size `max_len`.

`default_scatter_settings`: 
Calls the pbPlots function `GetDefaultScatterPlotSettings`, and adjusts the returned `ScatterPlotSetting` pointer to values fitting the project. The function takes a pointer to an already existing `ScatterPlotSeries` struct. Returns a pointer to the `ScatterPlotSettings` struct, which can be used as parameter to `DrawScatterPlotFromSettings`.

`free_scatter_settings`:
Takes a pointer to a `ScatterPlotSettings` struct, and frees all allocations in this struct, and also the struct itself.

`fill_data`:
Takes an `enum DataType`, a double pointer to an array, the size of the array, a `Datapoint` pointer to the CSV data, and a `struct tm`, which holds the users chosen day. The function fills the array with the data corresponding to the `DataType` parameter from the users chosen day onwards, untill the entire array has been filled.

### Changes in drawGraph.c
Changed functions to use the functions provided in graphUtils.h.

**New function `graph_comparison_scatter_exec`:**
Draws two scatterplots side by side :( . Takes an extra parameter compared to  `graph_scatterplot_exec` - a second type to graph. This second datatype is currently input in the switch-case in graph_exec, should maybe be changed if time allows.

Made wchar_t string arrays `type_strings` and `type_units` global static so all functions in drawGraph.c can access them.

### CODE REVIEW

File: PowerHouse/MenuFunctions.c [hunk 1]
```diff
@@ -66,8 +66,8 @@ void data_print_function(void)
 void graph_draw(void)
 {
     GraphParams input = graph_input();
-
-    graph_exec(input);
+
```
Comment (theis999) [inline PowerHouse/MenuFunctions.c:69]: Nice little whitespace

File: PowerHouse/MenuFunctions.c [hunk 2]
```diff
@@ -110,7 +110,7 @@ void calculate_appliance_run(void)
 
 //run 
     int total_rows;
-    Datapoint* data = GetCSVData(&total_rows);
+    Datapoint* data = readCSV("datafiler/DK-DK2_2022_hourly.csv", &total_rows, true);
```
Comment (theis999) [inline PowerHouse/MenuFunctions.c]: Why revert?

File: PowerHouse/MenuFunctions.c [hunk 3]
```diff
@@ -150,4 +150,7 @@ void calculate_appliance_run(void)
     {
         printf("The best time on %s is now\n", app.name);
     }
+
+    free(data);
```
Comment (theis999) [inline PowerHouse/MenuFunctions.c]: Linked to above comment

File: PowerHouse/drawGraph.c [hunk 4]
```diff
@@ -137,31 +76,100 @@ int graph_scatterplot_exec(DataType type, Datapoint *data, time_t day)
...
+
+    RGBABitmapImage *combined = CreateImage(640*2, 720, GetWhite());
+    RGBABitmapImage *image1, *image2;
+
+    ScatterPlotSettings *settings1 = GetDefaultScatterPlotSettings();
+    settings1->title = type_strings[type1];
+    settings1->titleLength = wcslen(settings1->title);
+    settings1->width = 640;
+    settings1->height = 720;
+    ScatterPlotSeries **s1 = malloc(sizeof(ScatterPlotSeries));
```
Comment (theis999) [inline PowerHouse/drawGraph.c:124]: I might have missed it, but i dont see where this is free'd

File: PowerHouse/drawGraph.c [hunk 5]
```diff
@@ -137,31 +76,100 @@ int graph_scatterplot_exec(DataType type, Datapoint *data, time_t day)
...
+    s1[0] = series1;
+    settings1->scatterPlotSeries = s1;
+    settings1->scatterPlotSeriesLength = 1;
+
+    ScatterPlotSettings *settings2 = GetDefaultScatterPlotSettings();
+    settings2->title = type_strings[type2];
+    settings2->titleLength = wcslen(settings2->title);
+    settings2->width = 640;
+    settings2->height = 720;
+    ScatterPlotSeries **s2 = malloc(sizeof(ScatterPlotSeries));
```
Comment (theis999) [inline PowerHouse/drawGraph.c:134]: I might have missed it, but i dont see where this is free'd

File: PowerHouse/graphUtils.c [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+{
+    // Calloc so string by default will be null-terminated.
+    char *temp = calloc(max_len+1, sizeof(char));
+    if (temp == NULL) {
+        return -1;
+    } 
+        
+    strftime(temp, max_len, format, timeptr);
+
+    size_t ret = mbstowcs(dest, temp, max_len);
```
Comment (theis999) [inline PowerHouse/graphUtils.c:38]: Freaking c naming convention...

**Review by theis999 — CHANGES_REQUESTED** [review]
Looks mostly fine. However without a good reason i'd prefer you kept using GetCSVData

File: PowerHouse/drawGraph.c (hunk 5 above)
Comment (chrnigi) [reply PowerHouse/drawGraph.c:134]: Freed with `free_scatter_settings` just before function return.

File: PowerHouse/drawGraph.c (hunk 4 above)
Comment (chrnigi) [reply PowerHouse/drawGraph.c:124]: Freed with `free_scatter_settings` just before function return.

File: PowerHouse/MenuFunctions.c (hunk 2 above)
Comment (chrnigi) [reply PowerHouse/MenuFunctions.c]: Not on purpose, happened during merge. Will un-revert.

File: PowerHouse/drawGraph.c (hunk 4 above)
Comment (theis999) [reply PowerHouse/drawGraph.c:124]: Fair enough

**Review by theis999 — APPROVED** [review]

**Pull request merged**